examples/moses-pln-synergy/scm/README.md

```diff
@@ -76,16 +76,204 @@ And search for the following
    )
 ...
 
-scheme@(guile-user)> (cog-bind pln-rule-eliminate-neutral-element-hack)
-scheme@(guile-user)> (cog-bind pln-rule-eliminate-dangling-junctor-hack)
-scheme@(guile-user)> (cog-bind pln-rule-equivalence-hack)
-$6 = (SetLink
+scheme@(guile-user)> (cog-bind implication-lambda-distribution-rule)
+...
    (ImplicationLink (stv 1 0.99999982)
-      (PredicateNode "take-treatment-1" (stv 0.1 0.80000001))
-      (PredicateNode "take-compound-A" (stv 0.2 0.80000001))
+      (LambdaLink
+         (TypedVariableLink
+            (VariableNode "$X")
+            (TypeNode "ConceptNode")
+         )
+         (AndLink
+            (EvaluationLink
+               (PredicateNode "take")
+               (ListLink
+                  (VariableNode "$X")
+                  (ConceptNode "treatment-1")
+               )
+            )
+            (EvaluationLink (stv 1 0.99999982)
+               (PredicateNode "contain")
+               (ListLink
+                  (ConceptNode "treatment-1")
+                  (ConceptNode "compound-A")
+               )
+            )
+         )
+      )
+      (LambdaLink
+         (TypedVariableLink
+            (VariableNode "$X")
+            (TypeNode "ConceptNode")
+         )
```

Comment:
Well, that's an interesting statement. Naively, in ordinary predicate logic, this would seem to be a fairly straight-forward thing to derive (not sure, I have not tried)  but here there is extra complexity, and I can't tell why the complexity is needed. You state that you want to use higher-order logic, implying that the variables could range over sets.  But then you convert everything to combinators, which have no free variables in them at all. -- that's not higher-order logic, that's just combinators.... I can't tell why this step is needed: it just seems to add complexity. 

Why not just use free variables, and a for-all link? Why all the fiddle-faddle with lambda?  What does the lambda allow you to do that ordinary predicate logic does not?